pipeline/data_store.py

```diff
@@ -926,43 +1006,80 @@ def get_reference(self, provenance=None, session=None):
         ref: Image object
             The reference image for this image, or None if no reference is found.
 
+
+        It will only return references whose validity date range
+        includes DataStore.image.observation_time.
+
+        If minovfrac is given, it will return the reference that has the
+        highest ovfrac.  (If, by unlikely chance, more than one have
+        identical overlap fractions, an undeterministically chosen
+        reference will be returned.  Ideally, by construction, you will
+        never have this situation in your database; you will only have a
+        single valid reference image for a given instrument/filter/date
+        that has an appreciable overlap with any possible image from
+        that instrument.  The software does not enforce this, however.)
```

Comment:
I think that since we are dealing with floating point, the only way this could happen is if both references use the exact same image. For images taken "of the same coordinates" the edges will be slightly different, if only because of rounding errors.

Reply:
Ideally, we won't have two different references defined that point to the same image, because then they'd be the same reference.  (In fact, should image_id be a "unique" field in the Reference table?  Hmm, probably not, because we might want to define the same image as a reference for multiple values of "target", for instance.)